src/lib/parsers/sol-json.c

```diff
@@ -289,28 +289,17 @@ sol_json_token_get_double(const struct sol_json_token *token, double *value)
      * of bytes.
      */
 
-    *value = sol_util_strtodn(token->start, &endptr,
+    errno = 0;
+    tmp_value = sol_util_strtodn(token->start, &endptr,
         sol_json_token_get_size(token), false);
+    SOL_INT_CHECK(errno, != 0, -errno);
```

Comment:
why do you want to change this? I tried to provide a "best value" in the case `strtod()` fails, like trying to set it to `-DBL_MAX` if -infinity.

Also, I recall having to set errno if value was `isinf()` or `isnan()`, could you confirm that still happens? Otherwise it would be returning unusable values with return 0.